internal/controller/organization/controller.go

```diff
@@ -0,0 +1,158 @@
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(o).
+		For(&v1alpha1.Organization{}).
+		Complete(r)
+}
+
+// A connector is expected to produce an ExternalClient when its Connect method
+// is called.
+type connector struct {
+	kube client.Client
+}
+
+// Connect typically produces an ExternalClient by:
+// 1. Tracking that the managed resource is using a ProviderConfig.
+// 2. Getting the managed resource's ProviderConfig.
+// 3. Getting the credentials specified by the ProviderConfig.
+// 4. Using the credentials to form a client.
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cl, err := clients.NewClient(ctx, c.kube, mg)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create a new client")
+	}
+	return &external{api: cl.OrganizationsAPI()}, nil
+}
+
+// An ExternalClient observes, then either creates, updates, or deletes an
+// external resource to ensure it reflects the managed resource's desired state.
+type external struct {
+	// A 'client' used to connect to the external resource API. In practice this
+	// would be something like an AWS SDK client.
+	api clients.OrganizationsAPI
+}
+
+func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.Organization)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotOrganization)
+	}
+
+	org, err := c.api.FindOrganizationByName(ctx, meta.GetExternalName(cr))
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(IsNotFound, err), errFindOrganization)
+	}
+
+	cr.Status.AtProvider = GetOrganizationObservation(org)
+	switch cr.Status.AtProvider.Status {
+	case "active":
+		cr.SetConditions(v1.Available())
+	case "inactive":
+		cr.SetConditions(v1.Unavailable())
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: pointer.StringDeref(org.Description, "") == pointer.StringDeref(cr.Spec.ForProvider.Description, ""),
+	}, nil
+}
+
+func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.Organization)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotOrganization)
+	}
+
+	_, err := c.api.CreateOrganization(ctx, &domain.Organization{
+		Description: cr.Spec.ForProvider.Description,
+		Name:        meta.GetExternalName(cr),
+	})
+
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreateOrganization)
+}
+
+func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.Organization)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotOrganization)
+	}
+
+	_, err := c.api.UpdateOrganization(ctx, &domain.Organization{
+		Name:        meta.GetExternalName(cr),
+		Description: cr.Spec.ForProvider.Description,
+		Id:          &cr.Status.AtProvider.ID,
```

Comment:
Using value from status subresource during update call? 🤔 
Do we have to do that? Could we find a better way to persist this information?

Reply:
We could possibly treat this as a resource whose external name is calculated but given that we're able to query by name and can make sure that we populate the ID in `Observe`, I opted for that pattern. But I don't really feel strongly.

Reply:
As we discussed offline, it should be ok given we could find and add it to status in observe.